### PR TITLE
Enemy behaviour

### PR DESCRIPTION
**Pull request by pkasinski**
Enemies now move toward the player character and push it back after colliding. Moreover, a generalized way of loading and getting entity images has been added, and a cyclops enemy type has been added.

### CODE REVIEW

File: core/src/model/Entity.java [hunk 1]
```diff
@@ -48,9 +55,6 @@ public void setMoving(boolean moving) {
         this.inMotion = moving;
     }
 
-    protected void setEntityName(String name){
```
Comment (rojtekin) [inline core/src/model/Entity.java:51]: Why is this removed?

File: core/src/model/ModelFactory.java [hunk 2]
```diff
@@ -1,9 +1,16 @@
 package model;
 
+import model.enemies.Cyclops;
```
Comment (rojtekin) [inline core/src/model/ModelFactory.java]: Should just import the whole enemies package

File: core/src/model/ModelFactory.java [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+import model.enemies.Mouse;
+
 public class ModelFactory {
     public static Model makeModel(String mapName) {
         Model model = new Model();
         model.initialize(new TiledMapLoader(mapName));
+        model.addEnemy(new Cyclops(100, 100, 1, 1, 1, model.getWorld()));
```
Comment (rojtekin) [inline core/src/model/ModelFactory.java:9]: Manually set into model i guess its only temporary, but if its temporary then why not do it in model? its gonna happen there the "correct way" anyway

File: core/src/model/PlacedMapEntity.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
  * A class for static objects placed on the map. Used to assign collision to map objects.
  */
 public class PlacedMapEntity extends Entity {
-    public PlacedMapEntity(float x, float y, float height, float width, World<Entity> world) {
-        super(x, y, height, width, 0, 1000000, world);
+    public PlacedMapEntity(float x, float y, float height, float width, World<Entity> world, String entityName) {
+        super(x, y, height, width, 0, 1000000, world, entityName);
```
Comment (rojtekin) [inline core/src/model/PlacedMapEntity.java]: Entityname should be a static attribute of every diffrent type of entity

File: core/src/model/PlayerCharacter.java [hunk 5]
```diff
@@ -5,7 +5,6 @@
 public class PlayerCharacter extends Entity implements IControllable {
 
     public PlayerCharacter(float spawnX, float spawnY, World<Entity> world) {
-        super(spawnX, spawnY, 32, 32, 5, 10, world);
-        setEntityName("BlueSamurai");
+        super(spawnX, spawnY, 32, 32, 5, 10, world, "Player");
```
Comment (rojtekin) [inline core/src/model/PlayerCharacter.java]: Entityname should be a static attribute of every diffrent type of entity

File: core/src/model/TiledMapLoader.java [hunk 6]
```diff
@@ -105,7 +105,7 @@ private void importMapCollision() {
         objects = map.getLayers().get(COLLISIONLAYER).getObjects();
         for (RectangleMapObject o : objects.getByType(RectangleMapObject.class)) {
             Rectangle r = o.getRectangle();
-            PlacedMapEntity st = new PlacedMapEntity(r.x, r.y, r.height, r.width, world);
+            PlacedMapEntity st = new PlacedMapEntity(r.x, r.y, r.height, r.width, world, "");
```
Comment (rojtekin) [inline core/src/model/TiledMapLoader.java]: this just looks wrong why is the last parameter ""

File: core/src/model/enemies/Cyclops.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package model.enemies;
+
+import model.Entity;
+import com.dongbat.jbump.World;
+
+public class Cyclops extends Enemy {
+    private boolean stuckInXDirection = false;
```
Comment (rojtekin) [inline core/src/model/enemies/Cyclops.java:7]: should be something applied to every non-player entity?

File: core/src/view/ImageHandler.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        return TextureRegion.split(spriteSheet,
+                spriteSheet.getWidth() / nColumnsPlayerWalkSheet,
+                spriteSheet.getHeight() / nRowsPlayerWalkSheet);
+    }
+
+    /**
+     * Loads the entity images and puts them in the HashMap allEntityImages.
+     */
+    protected void loadEntityImages() {
+        allEntityImages.put("Player", getEntityImages("Player"));
```
Comment (rojtekin) [inline core/src/view/ImageHandler.java]: repetitive code?

File: core/src/model/Entity.java (hunk 1 above)
Comment (pkasinski) [reply core/src/model/Entity.java:51]: Because the entity name does not need to be changed.

File: core/src/model/ModelFactory.java (hunk 3 above)
Comment (pkasinski) [reply core/src/model/ModelFactory.java:9]: Making the model is the responsibility of ModelFactory, and one aspect of making the model is adding the enemies.

**Review by AbdullahDab01 — APPROVED** [review]

**Review by rojtekin — APPROVED** [review]
LGTM
